infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/KubernetesInternalRuntime.java

```diff
@@ -210,9 +210,9 @@ protected void internalStart(Map<String, String> startOptions) throws Infrastruc
       final List<CompletableFuture<?>> toCancelFutures = new CopyOnWriteArrayList<>();
       final EnvironmentContext currentContext = EnvironmentContext.getCurrent();
       CompletableFuture<Void> startFailure = startSynchronizer.getStartFailure();
-
-      try (Scope waitRunningAsyncScope = tracer.buildSpan(WAIT_MACHINES_START).startActive(true)) {
-        TracingTags.WORKSPACE_ID.set(waitRunningAsyncScope.span(), workspaceId);
+      Span waitRunningAsyncSpan = tracer.buildSpan(WAIT_MACHINES_START).start();
```

Comment:
What is the difference here actually? I thought that `Scope.close()` which is being called after the try block, actually finishes the span?

Reply:
I have rewritten this class according to the SpanManger.activate javadoc.
```
     * <p>
     * Usage:
     * <pre><code>
     *     Span span = tracer.buildSpan("...").start();
     *     try (Scope scope = tracer.scopeManager().activate(span)) {
     *         span.setTag("...", "...");
     *         ...
     *     } catch (Exception e) {
     *         span.log(...);
     *     } finally {
     *         // Optionally finish the Span if the operation it represents
     *         // is logically completed at this point.
     *         span.finish();
     *     }
     * </code></pre>
     *
     * @param span the {@link Span} that should become the {@link #activeSpan()}
     * @return a {@link Scope} instance to control the end of the active period for the {@link Span}. It is a
     * programming error to neglect to call {@link Scope#close()} on the returned instance.
     */
    Scope activate(Span span);
```
Does it sounds like a good motivation?

Reply:
@metlos  I think the behavior you've mention was changed here https://github.com/opentracing/opentracing-java/pull/301

Reply:
:+1: Thanks!